### PR TITLE
configurable per env if haproxy or nginx is used for couch proxy

### PR DESCRIPTION
**Pull request by snopoke**
Once fully rolled out we can remove nginx references

Some conflicts with https://github.com/dimagi/commcare-cloud/pull/2207 which I'll resolve once that's merged

### CODE REVIEW

File: environments/development/public.yml [hunk 1]
```diff
@@ -44,6 +44,7 @@ nofile_limit: 65536
 
 supervisor_http_enabled: True
 
+couchdb_use_haproxy: True
```
Comment (emord) [inline environments/development/public.yml]: how long do you expect we'll be running nginx & haproxy?

Comment (snopoke) [reply environments/development/public.yml]: as long as it takes me to roll this out everywhere

Comment (snopoke) [reply environments/development/public.yml]: but hopefully not longer than a month

**Comment by snopoke** [comment]
having to rework this since the imports aren't playing nice

File: src/commcare_cloud/ansible/couchd2_nginx_cleanup.yml [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+
+    - name: remove nginx
+      package:
+        name: nginx
+        state: absent
+
+    - name: remove nginx config
+      file:
+        path: '/etc/nginx/'
+        state: abesent
```
Comment (emord) [inline src/commcare_cloud/ansible/couchd2_nginx_cleanup.yml]: spelling

File: src/commcare_cloud/ansible/couchd2_nginx_cleanup.yml [hunk 3]
```diff
@@ -0,0 +1,45 @@
+# Run this playbook once migration from nginx to haproxy is complete for couchdb2
```
Comment (emord) [inline src/commcare_cloud/ansible/couchd2_nginx_cleanup.yml]: this playbook is missing hte b in couchdb2

**Pull request merged**